federation/apis/federation/types.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package federation
+
+import (
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+)
+
+// Address of a cluster
+type ClusterAddress struct {
+	// URL to access the cluster
+	Url string `json:"url"`
+}
+
+// ClusterSpec describes the attributes on a Cluster.
+type ClusterSpec struct {
+	// a map of client CIDR to server address
+	ServerAddressByClientCIDRs []unversioned.ServerAddressByClientCIDR `json:"serverAddressByClientCIDRs"`
+	// The credential used to access cluster. It’s used for system routines (not behalf of users)
+	Credential string `json:"credential",omitempty`
```

Comment:
I dont think a string will do here.

Throwing out an idea (do not make any change yet):
What we finally want is to be able to populate [RestClient.Config](https://github.com/kubernetes/kubernetes/blob/1d96b4c55b6b4e6bc9bda5d12a7b99834056bdf4/pkg/client/restclient/config.go#L40) which is used by controller manager to get a new rest client (which the controller manager uses to talk to the apiserver).

The way it works right now is that it is populated from [clientcmd.Config](https://github.com/kubernetes/kubernetes/blob/b1aabc794dc2bb4908f68fd48ce258e5dc515d3f/pkg/client/unversioned/clientcmd/api/v1/types.go#L27) object (which is what kubeconfig file contains).
The relevant auth related fields there are:
```
InsecureSkipTLSVerify bool
// CertificateAuthority is the path to a cert file for the certificate authority.
CertificateAuthority string
// CertificateAuthorityData contains PEM-encoded certificate authority certificates. Overrides CertificateAuthority
CertificateAuthorityData []byte
```
Looking at my kubeconfig, it seems like we set InsecureSkipTLSVerify = false for local cluster and for clusters on GCE, we set CertificateAuthorityData.

One option is to add those fields here (maybe in a sub struct).